src/stringformat.c

```diff
@@ -82,18 +106,71 @@ GdipCreateStringFormat (INT formatAttributes, LANGID language, GpStringFormat **
 GpStatus WINGDIPAPI
 GdipStringFormatGetGenericDefault (GpStringFormat **format)
 {
-	return GdipCreateStringFormat (0, 0, format);
+	GpStatus status;
+
+	if (!format)
+		return InvalidParameter;
+
+	if (!gdiplusInitialized)
+		return GdiplusNotInitialized;
+
+#if GLIB_CHECK_VERSION(2,32,0)
+	g_mutex_lock (&generic);
+#else
+	g_static_mutex_lock (&generic);
+#endif
```

Comment:
Not a big fan of the mutexes. I wonder if it was easier to just allocate it during the library initialization and then simply return it here.

Reply:
Sure - this will have a perf/memory impact though but its tiny (like 20 bytes or something)